// compile with python -m grpc_tools.protoc -I ./ --python_out=. --grpc_python_out=. service.proto

syntax = "proto3";

service Worker {
    rpc ForwardMode(ForwardRequest) returns (ForwardReply) {}
    rpc BackwardMode(BackwardRequest) returns (BackwardReply) {}
    rpc ResetState(EmptyMessage) returns (EmptyMessage) {}
}

message EmptyMessage {}

// The request message containing the user's name.
message ForwardRequest {
    bool inference = 1;
    bytes system = 2; // pickle object
    string precision = 3;
    int32 n_frames = 4;
    string key = 5;
}

message ForwardReply {
    bytes du_dls = 1; // numpy array
    bytes energies = 2; // numpy array
    bytes frames = 3;
}

// The response message containing the greetings
message BackwardRequest {
    bytes adjoint_du_dls = 1; // pickle
    string key = 2;
}

message BackwardReply {
    bytes dl_dps = 1; // pickle
}